src/clj/rems/application/model.clj

```diff
@@ -503,31 +510,37 @@
                         days))
       application)))
 
+(defn- enrich-form-field-visible [form]
+  (let [field-values (build-index [:field/id] :field/value (:form/fields form))
+        update-field-visibility (fn [field] (assoc field :field/visible (field-visible? field field-values)))]
+    (transform [:form/fields ALL]
+               update-field-visibility
+               form)))
+
 (defn enrich-field-visible [application]
-  (let [fields (get-in application [:application/form :form/fields])
-        answers (into {} (map (juxt :field/id :field/value) fields))
-        fields-with-visible (mapv (fn [field]
-                                    (assoc field :field/visible (field-visible? field answers)))
-                                  fields)]
-    (-> application
-        (assoc-in [:application/form :form/fields] fields-with-visible))))
+  (transform [:application/forms ALL] enrich-form-field-visible application))
 
 (defn- enrich-disable-commands [application get-config]
   (permissions/blacklist application
                          (permissions/compile-rules
                           (for [command (:disable-commands (get-config))]
                             {:permission command}))))
 
-(defn enrich-with-injections [application {:keys [blacklisted?
-                                                  get-form-template get-catalogue-item get-license
-                                                  get-user get-users-with-role get-workflow
-                                                  get-attachments-for-application
-                                                  get-config]}]
-  (-> application
-      enrich-answers
-      (update :application/form enrich-form get-form-template)
-      enrich-field-visible ; uses enriched form fields and answers
-      set-application-description
+(defn enrich-with-injections
+  [application {:keys [blacklisted?
```

Comment:
for some reason I'd prefer a line return after `application`

Reply:
I think I prefer it like this. But I think we can refactor this code a bit later, if we do something about using `getx` or some `callx` here.

Reply:
ok